src/sagemaker/jumpstart/cache.py

```diff
@@ -339,11 +339,13 @@ def _retrieval_function(
                 formatted_content=model_specs
             )
         if data_type == HubDataType.MODEL:
-            hub_name, region, model_name, model_version = utils.extract_info_from_hub_content_arn(
+            info = utils.get_info_from_hub_resource_arn(
                 id_info
             )
-            hub = CuratedHub(hub_name=hub_name, region=region)
-            hub_content = hub.describe_model(model_name=model_name, model_version=model_version)
+            hub = CuratedHub(hub_name=info.hub_name, region=info.region)
```

Comment:
we really need to get rid of this class instantiation in this module. There should be a common utility/middleware to do this

Reply:
@jinyoung-lim is going to implement this, but I don't want to step on her toes yet